### PR TITLE
Docs: code server update

### PR DESCRIPTION
**Pull request by ZPain8464**
This PR optimizes the code-server guide and updates the content so it's current with Pomerium Core and Enterprise's latest versions. 

This PR resolves https://github.com/pomerium/internal/issues/1660 and is related to the Epic in https://github.com/pomerium/internal/issues/1622

### CODE REVIEW

**Review by desimone — APPROVED** [review]
Content LGTM; @cmo-pomerium on copy

File: content/docs/guides/code-server.mdx [hunk 1]
```diff
@@ -16,62 +16,66 @@ keywords:
...
 
-# Code-server
+# Secure Browser Access to code-server VSCode Behind Pomerium
+
+In this guide, you'll run code-server's Visual Studio Code (VSCode) in a Docker container, and secure browser access to your project with Pomerium.
 
 ## What is code-server?
 
-[Code-server](https://github.com/coder/code-server) is an open-source project that allows you to run [Visual Studio Code](https://code.visualstudio.com/) (VSCode) on a **remote server** through the browser.
+[Code-server](https://github.com/coder/code-server) is an open-source project that allows you to run [VSCode](https://code.visualstudio.com/) on a **remote server** through the browser.
```
Comment (cmo-pomerium) [inline content/docs/guides/code-server.mdx:31]:
```suggestion
[Code-server](https://github.com/coder/code-server) is an open-source project that allows you to run [VSCode](https://code.visualstudio.com/) on a **remote server** through the browser. VSCode is an open-source code editor by Microsoft that offers features like IntelliSense, Git integration, and plugins, while staying relatively lightweight.
```

File: content/docs/guides/code-server.mdx [hunk 2]
```diff
@@ -16,62 +16,66 @@ keywords:
...
 
 VSCode is an open-source code editor by Microsoft that offers features like IntelliSense, Git integration, and plugins, while staying relatively lightweight.
```
Comment (cmo-pomerium) [inline content/docs/guides/code-server.mdx:33]:
```suggestion
This setup essentially turns VS Code into a cloud-based IDE, providing flexibility and accessibility advantages by offering the full power of VS Code in a cloud-based environment. This is advantageous for those who work on multiple machines, need to access their development environment remotely, or have limited local resources.
```

**Review by cmo-pomerium — APPROVED** [review]
@ZPain8464 added my section to beef it up. Please review for copy (i did a quick rewrite)

**Pull request merged**